streams/src/main/java/org/apache/kafka/streams/state/internals/InMemoryWindowStore.java

```diff
@@ -48,7 +48,7 @@
 import static org.apache.kafka.streams.state.internals.WindowKeySchema.extractStoreKey;
 import static org.apache.kafka.streams.state.internals.WindowKeySchema.extractStoreTimestamp;
 
-public class InMemoryWindowStore<K, V> implements WindowStore<K, V> {
+public class InMemoryWindowStore<K extends Comparable<K>, V> implements WindowStore<K, V> {
```

Comment:
Temporary fix, will likely make the switch from generics to <Bytes, byte[]> in separate PR

Reply:
:+1: Maybe we should just consider that to be within the scope of https://issues.apache.org/jira/browse/KAFKA-7918 . (If you agree, can you leave a comment on the ticket to that effect?)